modules/diskq/qdisk.c

```diff
@@ -1440,8 +1440,13 @@ _load_qdisk_file(QDisk *self, const gchar *filename, GQueue *qout, GQueue *qback
 static gboolean
 _init_qdisk_file(QDisk *self)
 {
-  self->file_size = QDISK_RESERVED_SPACE;
-  return _create_header(self);
+  if (!_create_header(self))
+    return FALSE;
+
+  if (!self->file_size)
+    self->file_size = QDISK_RESERVED_SPACE;
```

Comment:
This (and the need for a detailed explanation) suggests that the placement of preallocation is not the best from a clean-code point of view.

`_create_file()` has the preallocation side-effect, which is a bit far from `_init_qdisk_file()`: it's in `_create_qdisk_file`.

I think this should look something like this:

```
_create_qdisk_file():
  _create_file();
  _create_header();
  if (!_preallocate())
    file_size = ...;
```

or

```
_create_qdisk_file():
  _create_file();
 
  if (need_prealloc)
    _prealloc();
  
  _create_header();

  if(!need_prealloc)
    file_size = ...
```

Or anything similar. I think the point still is that `_init_qdisk_file()` should not exist.

Reply:
Let me know if you don't want to fix this. We can merge it as-is.

Reply:
I agree, but if you don't mind I would do this in a new PR.

Reply:
Fixed in #4380.